url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgParseUsage.java

```diff
@@ -0,0 +1,754 @@
+        int baseCount = group.baseCount();
+        if (baseCount <= 1) {
+            argList.transfer(group);
+        }
+    }
+    
+    private void literal (String literal) {
+        
+        option = null;
+        
+        // Check for identical literal operand.
+        operand = findLit(literal);
+        
+        // Not found. Create a new one.
+        if (operand == null) {
+            operand = new ArgOperand().literal();
+            operand.setName(literal);
+            litList.add(operand);
+        }
+        if (exor) {
+            exor = false;
+            argList.appendSouth(operand);
+        } else {
+            argList.addEast(operand);
+        }
+        if (optionalDepth > 0) {
+            String token = source.next();
+            source.push();
+            if (token.equals("]")) {
+                ArgNode curr = argList.getCurrent();
+                ArgNode node = argList.endSouth();
+                node.setSpec(2);
+                argList.setCurrent(curr);
+            }
+        }
+    }
+    
+    private void literalOptions () {
+        ArgOperand operand = new ArgOperand().literal();
+        operand.setName("options");
+        operand.setOptional(true);
+        litList.add(operand);
+        argList.addEast(operand, true);
+    }
+    
+    private boolean parseOption () {
+        
+        // Next token.
+        String name = source.next();
+        if (name.equals("-")) {
+            name = fullName();
+        }
+        
+        // Single dash. Must be single letter option.
+        // Check validity.
+        else if (! validLetterOption(name)) {
+            return false;
+        }
+        
+        Debug.trace("parse option: " + name);
+        
+        // If not existing option, create a new one.
+        newOption = false;
+        option = findOption(name);
+        if (option == null) {
+            newOption = true;
+            option = new ArgOption(name);
+        }
+        return secondOption();
+    }
+    
+    private boolean secondOption () {
+        
+        // Next token not a comma. After option and description.
+        String token = source.next();
+        if (! token.equals(",")) {
+            source.push();
+            if (! afterOption()) {
+                return false;
+            }
+            if (newOption) {
+                addNewOption(1);
+            }
+            token = source.next();
+            if (token.equals("...") || token.equals("+")) {
+                option.setRepeat(true);
+                argList.setRepeat(true);
+                token = source.next();
+            }
+            return processDescription(token);
+        }
+        
+        // Set second option name.
+        if (! secondOptionName()) {
+            return false;
+        }
+        
+        // Description text.
+        token = source.next();
+        return processDescription(token);
+    }
+    
+    private boolean secondOptionName () {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 606-634 into a separate method.